src/compiler/checker.ts

```diff
@@ -19252,7 +19238,7 @@ namespace ts {
                     if (!(targetProp.flags & SymbolFlags.Prototype) && (!numericNamesOnly || isNumericLiteralName(name) || name === "length")) {
                         const sourceProp = getPropertyOfType(source, name);
                         if (sourceProp && sourceProp !== targetProp) {
-                            const related = propertyRelatedTo(source, target, sourceProp, targetProp, getTypeOfSymbol, reportErrors, intersectionState, relation === comparableRelation);
+                            const related = propertyRelatedTo(source, target, sourceProp, targetProp, getNonMissingTypeOfSymbol, reportErrors, intersectionState, relation === comparableRelation);

```

Comment:
@ahejlsberg question: did you miss a spot on line 19919 in `compareProperties`, i.e., should 
```typescript
compareTypes(getTypeOfSymbol(sourceProp), getTypeOfSymbol(targetProp));
```
have been replaced by 

```typescript
compareTypes(getNonMissingTypeOfSymbol(sourceProp), getNonMissingTypeOfSymbol(targetProp));
```
, thus preventing `{a?: string}` from comparing identical to `{a?: string | undefined}`?

Or was that intentional?

Reply:
This is a 5 year old PR; I would suggest commenting on an issue or filing a new one instead.

Reply:
@jakebailey thanks! I did comment on #61547 but since it was closed as "not a defect", I was hoping to understand very simply whether this behavior, when it was implemented, was intentional or non-intentional (to deduce whether the close reason was correct). If the behavior was _not_ intentional (which is my guess, since it seems very strange for an "identical" comparison to be _less strict_ than an assignment comparison), then that would provide evidence that the issue should be re-opened with this simple fix in mind. I could open a new issue but that seems strange when one already exists so it would effectively be a duplicate. Hope my reasoning makes sense here?